user guide for 1.0.7 pre-release features (#33)
exasol/powerbi-exasol: doc/user_guide/user_guide_107.md

> # Exasol Power BI Connector 1.0.7 Release - User Guide

Reviewer: why do you call it `107.md`. It versioned it git anyway.

> ## New features

Reviewer: Usually we don't include the new features here but in the changes_x.x.x.mde file.

> - Custom Query support

Reviewer: But the connector is also bundeled with power bi right?
Maybe you can tell here somewhere that it's just for the advanced users, and defualt users can pic the built-in connector

Author: it's for limited use

- ODBC Driver Logging

- DSN (Data Source Name) Support

- Limited Kerberos support

> ## Installing the connector:

Reviewer: Use assotiated press headline style: https://headlinecapitalization.com/

### Download the .mez

> Copy the ''Exasol.mez' file available on : [Releases · exasol/powerbi-exasol · GitHub](https://github.com/exasol/powerbi-exasol/releases)

Reviewer: ```suggestion
Download the 'Exasol.mez' file available on : [Releases · exasol/powerbi-exasol · GitHub](https://github.com/exasol/powerbi-exasol/releases)
```

You can find it separately under the assets dropdown.

This is the direct link:

https://github.com/exasol/powerbi-exasol/releases/download/1.0.7/Exasol.mez

### Copy the .mez to the following directory

Copy the mez file to `C:\Users\\<YourAccount>\Documents\Microsoft Power BI Desktop\Custom Connectors` Directory. 

Create the directory path if it doesn't exist.

### Change the PowerBI Desktop settings

Go to options in PowerBI:

File > Options and settings > Options

> ![](./img/2021-07-16-13-42-57-image.png)

Reviewer: :+1 nice to have images here

Reviewer: Could you rename them, with speaking names.

Reasons:
* Helps with screen readers
* Helps for search engine optimization

Navigate to the security tab.

Change the data extensions setting to allow the custom connector to load.

![](./img/2021-07-16-13-28-43-image.png)

> ### Shut down and restart Power BI Desktop

Reviewer: ```suggestion
### Restart Power BI Desktop
```

Let's call it "exa-test" and use the simplest connection string we can

![](./img/2021-07-16-14-06-10-image.png)

(You have an option to test the connection here to make sure it works. You might need to provide credentials but you can remove these after the test if you wish.) 

### Using the data source name in the connector

Write DSN=<your-data-source-name> as the connection string and press OK.

![](./img/2021-07-16-14-16-08-image.png)

Depending on whether you've provided credentials or not in your connection string you then pick one of the available authentication options and you should be succesfully connected.

#### Use case: ODBC Driver Logs

You can easily enable and disable ODBC driver logging using the data source options:

Navigate to the 'ODBC data sources' application.

Select your data source, click the "advanced" tab.

![](./img/2021-07-16-15-55-16-image.png)

You can easily turn logging on and off as well as select a desired log mode using the "Log file and "Log mode" fields.

Another option to do this is to manually add the relevant connection string keys to the connection string or to add them to Additional Connection string parameters under 'Advanced'.

> In this case that would be: LOGMODE=DEBUGCOMM;EXALOGFILE=C:\tmp\pbiblog.txt;

Reviewer: ```suggestion
In this case that would be: `LOGMODE=DEBUGCOMM;EXALOGFILE=C:\tmp\pbiblog.txt`;
```

Also add to other inline code snippets

#### Use case: Advanced Kerberos scenarios

In case the default built in Kerberos option isn't enough you can use a DSN and append KERBEROSHOSTNAME and KERBEROSSERVICENAME connection string keys and values to the connectionstring.

#### More information

There is also more documentation on exasol ODBC datasources configuration available here: [ODBC Driver for Windows | Exasol Documentation](https://docs.exasol.com/connect_exasol/drivers/odbc/odbc_windows.htm)

For an overview of the other available connection string keys, see this article:

[Using the ODBC Driver | Exasol Documentation](https://docs.exasol.com/connect_exasol/drivers/odbc/using_odbc.htm)

# Custom Query support

![](./img/2021-07-16-15-30-48-image.png)

It is now possible to write a custom query to fetch results.

You can alter this query afterwards in the Power Query editor as well.

Important note: You can only use a custom query in "import" mode. You can also use DirectQuery but won't be able to use the results in any visuals. 

(You can alter the mode afterwards in the model screen in case you accidentally picked the unsupported option.)

# Limited Kerberos support

> We also support an 'easy' kerberos authentication mode where the KERBEROSSERVICENAME will be exasol/<dnsrecord> if you choose the kerberos authentication option.

Reviewer: ``
```suggestion
We also support an 'easy' kerberos authentication mode where the `KERBEROSSERVICENAME` will be `exasol/<dnsrecord>` if you choose the kerberos authentication option.
```
